editor.planx.uk/src/pages/FlowEditor/components/Settings/Submissions/components/OpenResponseButton.tsx

```diff
@@ -0,0 +1,52 @@
+import Button from "@mui/material/Button";
+import CircularProgress from "@mui/material/CircularProgress";
+import { GridCellParams } from "@mui/x-data-grid";
+import React, { useState } from "react";
+import { DataTableModal } from "ui/shared/DataTable/components/DataTableModal";
+
+import { Submission } from "../types";
+import { FormattedResponse } from "./FormattedResponse";
+
+export const OpenResponseButton = (props: GridCellParams) => {
+  const { row } = props;
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+  const [response, setResponse] = useState<Record<string, any> | null>(null);
+
+  const getResponse = ({ eventType, status, response }: Submission) => {
+    if (eventType === "Pay") return response;
+    if (status === "Success") return response?.data?.body;
+
+    return response?.data?.message;
+  };
+
+  const handleButtonClick = () => {
+    setModalIsOpen(true);
+    if (!response) {
+      let fetchedData = getResponse(row);
+      try {
+        fetchedData =
+          typeof fetchedData === "string"
+            ? JSON.parse(fetchedData)
+            : fetchedData;
+      } catch (error) {
+        fetchedData = { error: "Invalid JSON format", raw: fetchedData };
+      }
+      setResponse(fetchedData);
+    }
+  };
+
+  return (
+    <>
+      <Button variant="text" size="small" onClick={handleButtonClick}>
+        View
+      </Button>
+      <DataTableModal
+        title="JSON response"
```

Comment:
nit: I wonder if we can avoid using "JSON" language with end-users - perhaps just "Response" or "Respose for [sessionId]" ??